nixos/tests/mysql-backup.nix

```diff
@@ -0,0 +1,61 @@
+# Test whether mysqlBackup option works
+import ./make-test.nix ({ pkgs, ... } : {
+  name = "mysql-backup";
+  meta = with pkgs.stdenv.lib.maintainers; {
+    maintainers = [ rvl ];
+  };
+
+  nodes = let
+    mkNode = { package, user }:
+      { config, pkgs, ... }: {
+        services.mysql = {
+          enable = true;
+          initialDatabases = [ { name = "testdb"; schema = ./testdb.sql; } ];
+          inherit package;
+        };
+
+        users.extraUsers.mybackup.isNormalUser = true;
+
+        services.mysqlBackup = {
+          enable = true;
+          databases = [ "doesnotexist" "testdb" ];
+          inherit user;
+        };
+      };
+  in {
+    one   = mkNode { package = pkgs.mysql;   user = "mysqlbackup"; };
+    two   = mkNode { package = pkgs.mariadb; user = "mysqlbackup"; };
+    three = mkNode { package = pkgs.mysql;   user = "mybackup"; };
+    four  = mkNode { package = pkgs.mariadb; user = "mybackup"; };
```

Comment:
Is their actually a usecase for having the username configureable?

Reply:
Good point. 🤔  What I can think of:

1. Someone creates mutliple instances of this (via custom derivations or something? Is that possible?) to have a different set of databases backed up with separate users to separate locations.

2. Someone backs up multiple things to `/var/backup` apart from databases and wants to use the same user for everything, to be able to take all what's there to another host via SSH. But in that case, I'd say it would be better to leave the `mysqlbackup` user as is, and have a general `backup` group to which all of `/var/backup` belongs to or something.

Especially in case it's not easily possible to have multiple instances, I'd prefer to see `mysqlBackup` as encapsulated service that always has that own `mysqlbackup` user which is only dumping databases to disk, and nothing else.

Reply:
You cannot do the first option. And the second option also don't require to change the username.

Reply:
Personally, I don't have a usecase for having the username configureable. However I'm reluctant to remove it because it was there before. What do you recommend?

In either case, considering that `mysql == mariadb` in nixpkgs, we could fairly safely collapse the four test cases into one to simplify and speed up the test.

Reply:
ok. then just remove the tests.

Reply:
OK, done, tests simplified.